Radio/index.jsx

```diff
@@ -225,14 +216,200 @@ const Radio = React.createClass({
             </div>
           ]
         })}
+
+        {collapsedOptions && <Collapsible
+          onStartFPSCollection={onStartFPSCollection}
+          onEndFPSCollection={onEndFPSCollection}
+          lowFPS={lowFPS}
+          minimumHeight={49}
+          collapsed={!isExpanded}>
+          <div>
+            {optionLists.collapsed.map((option) => {
+              const {
+                key,
+                label,
+                description,
+                disabled,
+                aside,
+                content,
+                leftPad,
+                ...restOfProps
+              } = option
+
+              const isDisabled = allDisabled || disabled
+              const id = `${name}-${key}`
+              const ids = {
+                aside: `${id}__aside`,
+                bullet: `${id}__bullet`,
+                checkmark: `${id}__checkmark`,
+                content: `${id}__content`,
+                description: `${id}__description`,
+                header: `${id}__header`,
+                headerInner: `${id}__header--inner`,
+                label: `${id}__label`,
+                labelInner: `${id}__label--inner`,
+                left: `${id}__left`,
+                right: `${id}__right`,
+                wrapper: `${id}__wrapper`
+              }
+
+              return [
+                <input
+                  className={classNames(classes.optionInput)}
+                  id={id}
+                  name={name}
+                  type='radio'
+                  onBlur={onBlur}
+                  checked={key === value}
+                  onChange={() => onChange && key !== value && onChange(key)}
+                  onFocus={(e) => onFocus && onFocus(key, e)}
+                  ref={key}
+                  value={key}
+                  disabled={isDisabled}
+                />,
+                <div
+                  className={classNames(
+                    classes.option,
+                    {
+                      'is-selected': key === value,
+                      'is-focused': !isDisabled && focus === key,
+                      'left-pad': leftPad && !singleOption,
+                      'is-disabled': isDisabled
+                    }
+                  )}
+                  id={ids.label}
+                  {...restOfProps}>
+                  <label
+                    htmlFor={`${name}-${key}`}
+                    className={classNames(classes.optionHeader)}
+                    id={ids.header}>
+                    <div
+                      className={classNames(classes.optionHeaderInner)}
+                      id={ids.headerInner}>
+                      {!singleOption && <div className={classNames(classes.optionLeft, classes.optionLeftmost)} id={ids.left}>
```

Comment:
This diff is green because I copy pasted. The decision was taken before. I know though it's a good opportunity for refactoring this "option" out, but I think that should also imply updating the styling strategy to inline (because in the current one the CSS is so tangled that extracting is hard). 

I don’t know what the deadline status is, maybe it's better to do it in a second pass.